tests/load/locustfiles/locustfile.py

```diff
@@ -31,119 +48,392 @@ def _(parser: Any):
+            return
 
-    def _time_event(self, name: str) -> TimeEvent:
-        return TimeEvent(self, name)
+        endpoint_url: str = random.choice(list(self.channels.values()))
+        self.post_notification(endpoint_url)
 
     def connect(self) -> None:
-        self.ws = create_connection(
+        """Creates the WebSocketApp that will run indefinitely."""
+        ws = websocket.WebSocketApp(
             self.environment.parsed_options.websocket_url,
-            header={"Origin": "http://localhost:1337"},
-            ssl=False,
+            header=self.WEBSOCKET_HEADERS,
+            on_message=self.on_ws_message,
+            on_error=self.on_ws_error,
+            on_close=self.on_ws_close,
+            on_open=self.on_ws_open,
         )
+        ws.run_forever()
 
-    def disconnect(self) -> None:
-        self.ws.close()
+    def post_notification(self, endpoint_url: str) -> None:
+        """Send a notification to Autopush.
 
-    def hello(self) -> None:
+        Args:
+            endpoint_url: A channel destination endpoint url
+        Raises:
+            ZeroStatusRequestError: In the event that Locust experiences a network issue while
+                                    sending a notification.
         """
-        Send a 'hello' message to Autopush.
+        message_type: str = "notification"
+        data: str = "TestData" + "".join(
+            [
+                random.choice(string.ascii_letters + string.digits)
+                for i in range(0, random.randrange(1024, 4096, 2) - 8)
+            ]
+        )
+
+        record = MessageRecord(message_type=message_type, send_time=time.perf_counter(), data=data)
+        self.message_records.append(record)
+
+        with self.client.post(
+            url=endpoint_url,
+            name=message_type,
+            data=data,
+            headers=self.REST_HEADERS,
+            catch_response=True,
+        ) as response:
+            if response.status_code == 0:
+                raise ZeroStatusRequestError()
+            if response.status_code != 201:
+                response.failure(f"{response.status_code=}, expected 201, {response.text=}")
+                return
+
+    def recv_hello(self, data: str, recv_time: float) -> None:
+        """Verify contents of 'hello' message and report response statistics to Locust.
+
+        Args:
+            data: utf-8 data received from the server
+            recv_time: Time when data was received
+        """
+        response_time: float = 0
+        exception: str | None = None
+
+        try:
+            message = HelloMessage(**json.loads(data))
+
+            # If a user is not subscribed to any channels upon re-connection a new
+            # ID will be assigned
+            self.uaid = message.uaid
+
+            # Get last 'hello' message record
+            record: MessageRecord = next(
+                (
+                    self.message_records[i]
```

Comment:
I'm a bit curious about this `message_records` bucket. It contains all messages and data regardless of the content? (e.g. `hello` responses mixed in with `notification` messages, etc.) We then do a scan of this bucket looking for a match, right? 

I'm wondering if we should divvy these up a bit so that notifications are distinct from commands that way we're not potentially missing if we cross streams at some point.

Reply:
We could do this, my original implementation had 3 queues.
The hello use case is wasteful for sure because I don't have a unique identifier and have to assume the last instance is correct.

message_records should be a a small data structure 30-40 entries max.